include/revng/TupleTree/TupleTree.h

```diff
@@ -66,6 +67,8 @@ class TupleTree {
         Reference = It->second;
     };
     visitReferences(Visitor);
+    if (ReferencesAreCached)
```

Comment:
1. Can be done
2. I agree in principle, but in practice it's tricky, because we have `visitReferences`, that is currently used
   - for initizalizing caches
   - for evicting caches
   - for verification
   - can be called by a user with a custom visitor
   
There are 2 versions of `visitReferences`: one `const` and one non-`const`. We'd like to put the assertion in the non-`const` but we cannot do it because it's used both to initialize and evict the cached targets. I can do a `private` version of the visit that does not assert and that is used only to initialize and evict the cached targets. What do you think?

Reply:
2. Sounds good

Reply:
Done